mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -52,16 +52,19 @@ public AccountManagerUtil(Context context, String accountName, boolean sharedSto
             //Attempt to retrieve the account
             Account[] accounts = mAccountManager.getAccountsByType(accountType);
             for (Account account : accounts) {
-                if (accountName.equals(account.name)) {
+                if (accountName != null &&accountName.equals(account.name)) {
                     String password = mAccountManager.getPassword(account);
                     String savedPassword = identifier.toString();
-                    if (password.equals(savedPassword)) {
+                    if (password != null && password.equals(savedPassword)) {
                         mAccount = account;
                     }else {
                         // - case migration from old AccountManagerStoreDataSource
                         mAccount = null;
                         identifier = new SharedStorageIdentifier();
+                        new NullPointerException("Password is null");
                     }
+                } else {
+                    new NullPointerException("Account Name does not exist");
```

Comment:
You forgot to throw the exception.
Use **_throw_** statement.

Reply:
And, it is not always meant to throw NullPointerException. 
Go for throwing **_IllegalArgumentException_** with an informative message.

Reply:
Looking at the below Ln:69, it seemed that you don't want to throw the exception if the password is missing.

Reply:
This is what I'm thinking:

if accountName is null, you can throw illegal argument exception
but if password is null, I think, you can continue without throwing the exception.

Because, mAccount is set to null, it will be initialized at the Ln:70.